core/types/src/keys.rs

```diff
@@ -334,3 +385,42 @@ where
         }
     }
 }
+
+/// Expose [`prost::Message`] for internal `KEY` types implementing this
```

Comment:
Nit: Trailing sentence. Also square brackets are treated as code links, no need for backticks
```suggestion
/// [Key] implements [prost::Message] when its `KEY` type implements [prost::Message].
```